src/test/java/com/exasol/errorreporting/QuoterTest.java

```diff
@@ -20,14 +28,87 @@ static Stream<Arguments> getQuotingExamples() {
                 Arguments.of(List.of(1, "test"), "[1, 'test']"), //
                 Arguments.of(new CustomObject(), "<CustomObject>"), //
                 Arguments.of(null, "<null>"), //
-                Arguments.of("test", "'test'")//
+                Arguments.of("test", "'test'"), //
+                Arguments.of(new File(SEP + "foo" + SEP + "baz.txt"), "'"+ SEP + "foo" + SEP + "baz.txt'"), //
+                Arguments.of(Path.of("foo").resolve("bar"), "'foo" + SEP + "bar'"), //
+                Arguments.of(new URL("https://example.org"), "'https://example.org'"), //
+                Arguments.of(new URI("URN:ISBN:0-330-28700-1"), "'URN:ISBN:0-330-28700-1'")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("getAutoQuotingExamples")
+    void testAutoQuoting(final Object input, final String expectedOutput) {
+        assertThat(Quoter.quoteObject(input, Quoting.AUTOMATIC), equalTo(expectedOutput));
+    }
+
+    static Stream<Arguments> getUnquotedExamples() throws MalformedURLException, URISyntaxException {
+        return Stream.of(//
+                Arguments.of(1, "1"), //
+                Arguments.of(1L, "1"), //
+                Arguments.of(1f, "1.0"), //
+                Arguments.of(1d, "1.0"), //
+                Arguments.of(List.of(1, "test"), "[1, test]"), //
+                Arguments.of(new CustomObject(), "<CustomObject>"), //
+                Arguments.of(null, "<null>"), //
+                Arguments.of("test", "test"), //
+                Arguments.of(new File(SEP + "foo" + SEP + "baz.txt"), SEP + "foo" + SEP + "baz.txt"), //
```

Comment:
This test is platform dependent.
```suggestion
                Arguments.of(new File("/foo/baz.txt"), "/foo/baz.txt"), //
                Arguments.of(new File("\\foo\\baz.txt"), "\\foo\\baz.txt"), //
```
(same for the following tests)